charts/k8s-monitoring/templates/agent_config/_agent.river.txt

```diff
@@ -1,41 +1,77 @@
 {{ define "agent.config.agent" }}
 // Grafana Agent
 discovery.relabel "agent" {
-  targets = discovery.kubernetes.services.targets
+  targets = discovery.kubernetes.pods.targets
   rule {
-    source_labels = ["__meta_kubernetes_service_label_app_kubernetes_io_instance"]
+    source_labels = ["__meta_kubernetes_pod_label_app_kubernetes_io_instance"]
     regex = "{{ .Release.Name }}"
     action = "keep"
   }
+  {{- range $k, $v := .Values.metrics.agent.labelMatchers }}
   rule {
-    source_labels = ["__meta_kubernetes_service_label_app_kubernetes_io_name"]
-    regex = "grafana-agent"
+    source_labels = ["__meta_kubernetes_pod_label_{{ $k | replace "." "_" | replace "/" "_" }}"]
+    regex = "{{ $v }}"
     action = "keep"
   }
+  {{- end }}
   rule {
-    source_labels = ["__meta_kubernetes_service_port_name"]
-    regex = "http-metrics"
+    source_labels = ["__meta_kubernetes_pod_container_port_name"]
+    regex = "{{- .Values.metrics.agent.pod.port -}}"
     action = "keep"
   }
+  rule {
+    source_labels = ["__meta_kubernetes_namespace"]
+    target_label  = "namespace"
+  }
+  rule {
+    source_labels = ["__meta_kubernetes_pod_name"]
+    target_label  = "pod"
+  }
+  rule {
+    source_labels = ["__meta_kubernetes_pod_container_name"]
+    target_label  = "container"
+  }
+{{- if .Values.metrics.extraRelabelingRules }}
+{{ .Values.metrics.extraRelabelingRules | indent 2 }}
+{{- end }}
+{{- if .Values.metrics.agent.extraRelabelingRules }}
+{{ .Values.metrics.agent.extraRelabelingRules | indent 2 }}
+{{- end }}
 }
 
 prometheus.scrape "agent" {
-  job_name   = "integrations/agent"
-  targets  = discovery.relabel.agent.output
+  job_name = "integrations/agent"
+  targets = discovery.relabel.agent.output
+  scrape_interval = {{ .Values.metrics.agent.scrapeInterval | default .Values.metrics.scrapeInterval | quote }}
   forward_to = [prometheus.relabel.agent.receiver]
 {{- if (index .Values "grafana-agent").agent.clustering.enabled }}
   clustering {
     enabled = true
   }
 {{- end }}
+{{- if .Values.metrics.agent.pod.isTLS }}
+  scheme = "https"
+  bearer_token_file = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+  tls_config {
+    insecure_skip_verify = true
+  }
+{{- end }}
```

Comment:
Remove this. We haven't needed TLS for the agent, so let's not complicated it until its required.